tests/test_protocol.py

```diff
@@ -1700,3 +1702,282 @@ async def merge_dicts_impl(self, one: Dict[str, Any], other: Dict[str, int], any
     response: Result = await client.merge_dicts(one, other, None)
     assert response.code == 200
     assert response.result == {"data": {**one, **other}}
+
+
+@pytest.mark.parametrize(
+    "param_type,param_value,expected_url,expected_return_value, should_unquote",
+    [
+        (Dict[str, str], {"a": "b", "c": "d"}, "/api/v1/test/1/monty?filter.a=b&filter.c=d", "a,c", False),
+        (Dict[str, List[str]], {"a": ["b"], "c": ["d", "e"]}, "/api/v1/test/1/monty?filter.a=b&filter.c=d,e", "a,c", True),
```

Comment:
could you add a key and a value to the test that contain ',&?=% '

Reply:
I extended the test case, and it doesn't seem to cause any issues, except introducing an empty string into the lists (which can also be done on purpose). I don't think it's very useful to send empty strings to an API, but it might not be necessarily wrong either. Should we consider this a `BadRequest` or filter them out before passing the values to the handler, or just leave them in?

Reply:
I don't understand about the empty list.

Perhaps the testcase would be easier to understand if the service just returns the parameter value it receives, such that 
`param_value == expected_return_value` 

This way, it is clear that it cleanly round-trips?

Reply:
Ok, I see, you can disregard my previous comment